timeboost/s3_storage.go

```diff
@@ -112,13 +113,27 @@ func (s *S3StorageService) Start(ctx context.Context) {
 	}
 }
 
-func (s *S3StorageService) uploadBatch(ctx context.Context, batch []byte, fistRound uint64) error {
+// Used in padding round numbers to a fixed length for naming the batch being uploaded to s3. <firstRound>-<lastRound>
+const fixedRoundStrLen = 7
+
+func (s *S3StorageService) getBatchName(firstRound, lastRound uint64) string {
+	padRound := func(round uint64) string {
```

Comment:
You can just use string formatting for this 

```
package main

import (
	"fmt"
	"strconv"
)

const fixedRoundStrLen = 7

func main() {
	fmt.Printf("Hello, %0"+strconv.Itoa(fixedRoundStrLen)+"d", 1)
}
```

prints 
```
Hello, 0000001
```

Reply:
Ah yess, thanks!